offscreen-canvas/the-offscreen-canvas/offscreencanvas.commit.html

```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/canvas-tests.js"></script>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/scripting.html#the-offscreen-2d-rendering-context">
+<script>
+
+function examPlaceholder(placeholder)
+{
+    var canvas = document.createElement('canvas');
+    canvas.width = canvas.height = 10;
+    var ctx = canvas.getContext('2d');
+    ctx.drawImage(placeholder, 0, 0);
+    _assertPixel(canvas, 5,5, 0,255,0,255, "5,5", "0,255,0,255");
+}
+
+test(function() {
+    var placeholder = document.createElement('canvas');
+    placeholder.width = placeholder.height = 10;
+    var offscreenCanvas = placeholder.transferControlToOffscreen();
+    var ctx = offscreenCanvas.getContext('2d');
+    ctx.fillStyle = "#0f0";
+    ctx.fillRect(0, 0, 10, 10);
+    ctx.commit();
```

Comment:
You should also verify that the placeholder is still blank immediately after the commit.  Commit() propagation is taken care of by an async task (as per the spec), which means that placeholder contents are never updated before the end of the JS task that called commit in cases where commit is called from the same event loop where the placeholder lives.